File: csweb2.0-RainmakersAWS.12/csweb2.0-client/src/app/core/interfaces/aob3-project-details.ts
export interface ActivityHistory {
  step: number;
  activity: string;
  status: string;
  startTime: string;
  endTime: string;
}

export interface Aob3ProjectDetails {
  projectName: string;
  startDate: string;
  projectStatus: string;
  completionDate: string;
  noOfAccounts: number;
  noOfBillTos: number;
  noOfSites: number;
  noOfContainers: number;
  noOfRates: number;
  noOfRoutes: number;
  currentStep: string;
  currentStepStatus: string;
  error: string;
  lastAttempt: string;
  lastValidated: string;
  nextStep: string;
  activityhistory: ActivityHistory[];
}

File: csweb2.0-RainmakersAWS.12/csweb2.0-api/src/main/java/com/repsrv/csweb/core/model/account/aob3/Aob3ProjectDetails.java
package com.repsrv.csweb.core.model.account.aob3;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Aob3ProjectDetails {

    private String projectName;

    private String startDate;

    private String projectStatus;

    private String completionDate;

    private int noOfAccounts;

    private int noOfBillTos;

    private int noOfSites;

    private int noOfContainers;

    private int noOfRates;

    private int noOfRoutes;

    private String currentStep;

    private String currentStepStatus;

    private String error;

    private String lastAttempt;

    private String lastValidated;

    private String nextStep;

    @SerializedName("activityhistory")
    private List<ActivityHistory> activityHistory;
    
}
